### PR TITLE
EUI-3045: Merged into performance improvements code

### PR DESCRIPTION
**Pull request by paul-graham**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/EUI-3045

### Change description ###
@phillip-whitaker created a specific dynamic read/write dropdown field. The new write dropdown component checks whether a value property has already been set from the API and if not, sets the value to null so that the default "-- Select an option --" is shown rather than a blank dropdown. This has now been merged into the performance improvements branch with the requisite changes to retain the functionality.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Review by phillip-whitaker — APPROVED** [review]

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@hmcts/ccd-case-ui-toolkit",
-  "version": "2.71.0-conditional-show-perf",
+  "version": "2.71.4-optional-top-level-documents",
```
Comment (paul-graham) [inline package.json:3]: The latest on `master` is `2.72.1-something-or-other`.

File: src/shared/components/palette/dynamic-list/read-dynamic-list-field.component.spec.ts [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+    it('render null value as empty string', () => {
+      component.caseField.value = null;
+      fixture.detectChanges();
+
+      expect(de.nativeElement.textContent).toEqual(EMPTY);
+    });
+  });
+
+  describe('ReadDynamicListFieldComponent for dynamiclist type', () => {
+    const VALUE_DYNAMIC_LIST = '{\n' +
```
Comment (paul-graham) [inline src/shared/components/palette/dynamic-list/read-dynamic-list-field.component.spec.ts:112]: A string literal would be much easier to read and avoid the need for the `\n`s entirely.

**Review by paul-graham — COMMENTED** [review]
Apparently, I can't request changes on my own PR (makes sense) so I'm just commenting. :)

File: demo/src/app/app.config.ts [hunk 3]
```diff
@@ -139,4 +139,8 @@ export class AppConfig extends AbstractAppConfig {
   public getWorkAllocationApiUrl(): string {
     return this.config.work_allocation_api_url;
   }
+
+  public getUserInfoApiUrl(): string {
```
Comment (paul-graham) [inline demo/src/app/app.config.ts:143]: Really not understanding why this is an addition... it's already on `master`, isn't it?

**Comment by paul-graham** [comment]
Replaced by https://github.com/hmcts/ccd-case-ui-toolkit/pull/697.

**Pull request closed without merging**